elyra/util/archive.py

```diff
@@ -55,19 +55,18 @@ def tar_filter(tarinfo):
         if '*' in files:
             return tarinfo
 
-        # if tarinfo.name == os.path.basename(operation.artifact):
-        #    return tarinfo
-
         for dependency in files:
             if dependency:
                 if dependency.startswith('*'):
                     # handle check for extension wildcard
                     if tarinfo.name.endswith(dependency.replace('*.', '.')):
                         return tarinfo
-                else:
+                elif tarinfo.name == dependency:
                     # handle check for specific file
-                    if tarinfo.name == dependency:
-                        return tarinfo
+                    return tarinfo
+                elif recursive:
```

Comment:
where is `recursive` coming from? or is it some python keyword I just haven't heard of

Reply:
nvm, I should read adjacent code closer next time